handlers/common.go

```diff
@@ -189,6 +191,56 @@ func initEnviron(machineDir string) []string {
 	return env
 }
 
+func reinitFromExtractedConfig(machine *client.Machine, machineBaseDir string) error {
+	if err := os.MkdirAll(machineBaseDir, 0740); err != nil {
+		return fmt.Errorf("Error reinitializing config (MkdirAll). Config Dir: %v. Error: %v", machineBaseDir, err)
+	}
+
+	configBytes, err := b64.StdEncoding.DecodeString(machine.ExtractedConfig)
+	if err != nil {
+		return fmt.Errorf("Error reinitializing config (base64.DecodeString). Config Dir: %v. Error: %v", machineBaseDir, err)
+	}
+
+	gzipReader, err := gzip.NewReader(bytes.NewReader(configBytes))
+	if err != nil {
+		return err
+	}
+	tarReader := tar.NewReader(gzipReader)
+
+	for {
+		header, err := tarReader.Next()
+		if err != nil {
+			if err == io.EOF {
+				return nil
+			}
+			return fmt.Errorf("Error reinitializing config (tarRead.Next). Config Dir: %v. Error: %v", machineBaseDir, err)
+		}
+
+		filename := header.Name
+		filePath := filepath.Join(machineBaseDir, filename)
+		log.Infof("Extracting %v", filePath)
+
+		info := header.FileInfo()
+		if info.IsDir() {
+			err = os.MkdirAll(filePath, os.FileMode(header.Mode))
+			if err != nil {
+				return fmt.Errorf("Error reinitializing config (Mkdirall). Config Dir: %v. Dir: %v. Error: %v", machineBaseDir, info.Name(), err)
+			}
+			continue
+		}
+
+		file, err := os.OpenFile(filePath, os.O_CREATE|os.O_TRUNC|os.O_WRONLY, info.Mode())
```

Comment:
Yeah, I figured since I was creating the archive, it was ok.

But FWIW, I also changed the structure of the archive, so this HA code won't work for "legacy" machines. My reasoning here was that it was not worth the extra effort and added complexity of the code to handle the old format. Also, the legacy format doesn't include the certs that are now necessary to get a docker client for the machine.